experimental/PlaneCurve/AffineEllipticCurve.jl

```diff
@@ -0,0 +1,74 @@
+export AffineEllipticCurve, discriminant, issmooth, j_invariant
+
+################################################################################
+# Helping functions
+################################################################################
+
+function isweierstrassform(eq::Oscar.MPolyElem{S}) where S <: FieldElem
+   R = parent(eq)
+   x = gen(R, 1)
+   y = gen(R, 2)
+   A = [x*y, y, x^2, x, R(1)]
+   f = deepcopy(eq)
+   f = f - y^2 + x^3
+   for t in A
+      f = f - coeff(eq, t)*t
+   end
+   iszero(f) && return true
+   return false
+end
+
+function shortformtest(eq::Oscar.MPolyElem{S}) where S <: FieldElem
+   R = parent(eq)
+   x = gen(R, 1)
+   y = gen(R, 2)
+   A = [coeff(eq, x*y), -coeff(eq, x^2), coeff(eq, y), -coeff(eq, x), -coeff(eq, R(1))]
+   iszero(A[1]) && iszero(A[2]) && iszero(A[3]) && return (true, [A[4], A[5]])
+   return (false, A)
+end
+
+################################################################################
+# Definition EllipticCurve
+################################################################################
+
+mutable struct AffineEllipticCurve{S <: FieldElem} <: PlaneCurve
```

Comment:
Stupid question, but isn't an AffineEllipticCurve a special case of an AffinePlaneCurve? Or perhaps it is not (e.g. because you want to "include the point at infinity") -- then perhaps a comment could state that and point out the reason?

(I am happy to merge this as-is and you address this in a future PR, I just wonder)

Reply:
It's a preliminary structure for elliptic curves, and it will change in future pull requests. I am not sure which shape I will need. I might change them so that they are projective, but it is for the moment easier to handle in the affine case.

Reply:
fine, thanks!